### ISSUE TITLE
Version 5.0.1 opens in Russian whatever the setting (MacOS)

### ISSUE THREAD

**Issue by bouscram** [sonic-visualiser/sonic-visualiser#102]
My Mac (running MacOS 15.1) is set to French, and to my surprise version 5.0.1 opened in Russian.
I tried to change the setting (in the System parameters), but whatever the setting, (English, Czech and - obviously - Russian), the program uses Russian language.

**Comment by cannam** [comment]
Can you check your Mac language settings (System Settings -> General -> Language & Region -> Preferred Languages - or the equivalent in French of course!) and verify that you do not have Russian listed as an accepted language?

SV has (user-contributed) translations for only Czech and Russian (as well as English). So if your system has (say) French and Russian configured in that order, it will come up in Russian even though you had French listed first, because there is no translation for French.

The reason I ask you to check this, is that two other users have reported the same thing and in both cases it turned out that they had the system language preferences set to German and then Russian (and they had either forgotten this, or never realised) so it came up in Russian because there was no German translation.

Although this behaviour is technically correct, I don't think it's desirable, and I do want to change it - I think for most users it would be less surprising it if used only the primary language and then fell back to English. But I would like to find out whether this is the situation you are seeing, or whether there is another cause for the problem in your case.

**Comment by cannam** [comment]
(Incidentally, one of the reports I mentioned was for 5.0.0, and one of the reasons I made the 5.0.1 release was to try to fix this - but the fix didn't work correctly, because I misunderstood an aspect of how Qt locale support works)

**Comment by bouscram** [comment]
Thanks for answering quickly. On my machine, languages are in the following order : French / English / Russian, so it should display English. What I tested initially was to change the application-level settings (at the bottom of the same pane). I must add that it works correctly for 5.0.0 on my Mac.

Removing Russian from the systemwide setting makes SV 5.0.1 open correctly in English. An interesting side effect is that now if I select Czech as application language, it displays Russian... :-/

**Comment by cannam** [comment]
Thanks for the extra info. That's interesting about English being listed but not used. I wonder whether the fact that the English translation (at least en_US) is built-in to SV rather than being loaded externally makes a difference.

**Comment by bouscram** [comment]
I found something interesting: if I add English (US) to the list of system languages (before Russian), SV opens in English. I guess that the language setting in the app is too restrictive...

**Comment by akawshi** [comment]
I had similar problem with Windows for the version 5.2.1. I also have Russian installed as a language, but I am not a Russian speaker, as I have it to produce Cyrillic characters.

My native language is Japanese, but perhaps it was skipped because there is no Japanese in this software. Do I have to delete the Russian data from Windows and reinstall SonicVisualizer to use it in English?

**Comment by cannam** [comment]
> My native language is Japanese, but perhaps it was skipped because there is no Japanese in this software. Do I have to delete the Russian data from Windows and reinstall SonicVisualizer to use it in English?

You are correct that Sonic Visualiser doesn't have a Japanese translation and so the system preference for Japanese will (sadly) be ignored completely.

SV should instead pick up whichever of its supported languages appears first in the system configuration. So:

 * If you have both Russian and English (any variant) installed in the system, and English is listed with a higher system priority - e.g. the system languages are Japanese, English, Russian - then it should appear in English
 * If you have both Russian and English installed but Russian has a higher priority - e.g. the system languages are Japanese, Russian, English - then it should appear in Russian
 * If you have only one of English or Russian installed, then it should use that
 * If you have none of the supported languages installed, then it should appear in English

Any other behaviour might be a bug.

**Comment by akawshi** [comment]
Thank you for your help. I changed language preference and could set SV display to English.
